independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/converters/OptionalConverter.java

```diff
@@ -18,7 +19,14 @@ public Object convert(Object parameter) {
         if (parameter == null) {
             return Optional.empty();
         } else if (delegate != null) {
-            return Optional.ofNullable(delegate.convert(parameter));
+            Object converted = delegate.convert(parameter);
+            if (converted != null
+                    && converted instanceof Collection
+                    && ((Collection) converted).isEmpty()) {
```

Comment:
Is it really necessary to check if the collection is Empty ?

Reply:
To return optional empty when the collection is empty, yes. See my comment here: https://github.com/quarkusio/quarkus/pull/24055#issuecomment-1056810692

Reply:
Oh I see, you are right, It's better this way